cores/esp32/IPAddress.h

```diff
@@ -1,114 +1,116 @@
 /*
- IPAddress.h - Base class that provides IPAddress
- Copyright (c) 2011 Adrian McEwen.  All right reserved.
+  IPAddress.h - Base class that provides IPAddress
+  Copyright (c) 2011 Adrian McEwen.  All right reserved.
 
- This library is free software; you can redistribute it and/or
- modify it under the terms of the GNU Lesser General Public
- License as published by the Free Software Foundation; either
- version 2.1 of the License, or (at your option) any later version.
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
 
- This library is distributed in the hope that it will be useful,
- but WITHOUT ANY WARRANTY; without even the implied warranty of
- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- Lesser General Public License for more details.
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
 
- You should have received a copy of the GNU Lesser General Public
- License along with this library; if not, write to the Free Software
- Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
- */
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
 
-#ifndef IPAddress_h
-#define IPAddress_h
+#pragma once
 
 #include <stdint.h>
-#include <WString.h>
-#include <Printable.h>
-
-// A class to make it easier to handle and pass around IP addresses
+#include "Printable.h"
+#include "WString.h"
+#include "lwip/ip_addr.h"
 
 #define IPADDRESS_V4_BYTES_INDEX 12
 #define IPADDRESS_V4_DWORD_INDEX 3
 
-enum IPType
-{
+// A class to make it easier to handle and pass around IP addresses
+
+enum IPType {
     IPv4,
     IPv6
 };
 
-class IPAddress: public Printable
-{
+class IPAddress : public Printable {
 private:
     union {
         uint8_t bytes[16];
```

Comment:
There was a discussion (and another PR... or maybe it was in ArduinoCore) about removing the union, as technically conversions between the different types in a union is undefined. Just have a single uint8_t bytes[16] array, and then cast as necessary. (although I'm pretty sure the standard C socket interface uses a very similar union).

Anyway, I don't think the explicit union adds a lot of value, especially as we are mostly just accessing it through Arduino methods.

Reply:
I'm also not a fan of union, but how it is used here, you also get the IPv4-as-IPv6-address implementation for free.  (unless I'm making the same mistake as always where I mess up the bit order in my mind, exactly the reason why I'm not a fan of unions but still use them myself...)

Reply:
This whole class (both header and implementation) I copied directly from Arduino's Core API. Anything that is not `to/from_ip_addr_t` and `_zone` is a carbon copy of the official code.

Reply:
There is an effort to change the approach in Arduino, if that happens so, we will update the class here too.